### PR TITLE
move setting route handlers to registration from start

### PR DESCRIPTION
**Pull request by james-prysm**
**What type of PR is this?**

 Bug fix

**What does this PR do? Why is it needed?**

There is a race introduced by registering the route at the start of the services within the beacon chain particularly between the rpc service and the grpc gateway service which use the underlying route. under these conditions if the enduser calls an endpoint without a pause at the start of the service a panic may occur. This PR moves the setting of route handlers in the registration instead of when the service starts.

example of an error 
```
2024/02/23 09:24:09 http: panic serving 172.18.0.6:59022: runtime error: invalid memory address or nil pointer dereference
goroutine 214271 [running]:
net/http.(*conn).serve.func1()
	GOROOT/src/net/http/server.go:1868 +0xb9
panic({0x42c7e0?, 0x33f1ee0?})
	GOROOT/src/runtime/panic.go:920 +0x270
github.com/gorilla/mux.(*Route).Match(0xc003ad4460?, 0x0?, 0xc00f08c420?)
	external/com_github_gorilla_mux/route.go:42 +0x1a
github.com/gorilla/mux.(*Router).Match(0xc0001d8240, 0x1?, 0xc00f08c420)
	external/com_github_gorilla_mux/mux.go:138 +0x6b
github.com/gorilla/mux.(*Router).ServeHTTP(0xc0001d8240, {0xd109e0, 0xc009229a40}, 0xc006a43900)
	external/com_github_gorilla_mux/mux.go:196 +0xd9
github.com/rs/cors.(*Cors).Handler-fm.(*Cors).Handler.func1({0xd109e0, 0xc009229a40}, 0xc006a43900)
	external/com_github_rs_cors/cors.go:219 +0x17e
net/http.HandlerFunc.ServeHTTP(0x3788780?, {0xd109e0?, 0xc009229a40?}, 0xc00238fb50?)
	GOROOT/src/net/http/server.go:2136 +0x29
net/http.serverHandler.ServeHTTP({0xc00f08c240?}, {0xd109e0?, 0xc009229a40?}, 0x6?)
	GOROOT/src/net/http/server.go:2938 +0x8e
```

### CODE REVIEW

File: beacon-chain/rpc/service.go [hunk 1]
```diff
@@ -557,8 +425,6 @@ func (s *Service) Start() {
 	// Register reflection service on gRPC server.
 	reflection.Register(s.grpcServer)
 
-	validatorServer.PruneBlobsBundleCacheRoutine()
```
Comment (nisdas) [inline beacon-chain/rpc/service.go:560]: why is this moved ?

Comment (james-prysm) [reply beacon-chain/rpc/service.go:560]: it's not, it's part of the Start() function still but the validator server needed to be set somewhere, i moved all of the initialization functions out of start and introduced the validator server as another variable to be used.

see line 587 s.validatorServer.PruneBlobsBundleCacheRoutine()

**Review by rkapka — APPROVED** [review]
